src/icq/client/events.rs

```diff
@@ -0,0 +1,104 @@
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Event {
+    pub seq_num: u32,
+
+    #[serde(flatten)]
+    pub event_data: EventData,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(tag = "type", content = "eventData")]
+#[serde(rename_all = "camelCase")]
+pub enum EventData {
+    #[serde(rename_all = "camelCase")]
+    HistDlgState {
```

Comment:
`HistDlgState(HistDlgStateData)` didn't work?

Reply:
I didn't try. What are the advantages?

Reply:
Would allow doing:
```rust
match event_data {
  HistDlgState(data) => process_hist_dlg_state(data),
  ...
}
```

Reply:
Isn't that already what we do in poller.rs?

Reply:
No, 

```
match event.event_data {
    EventData::HistDlgState { sn, starting, last_msg_id, last_read_mention, .. } => ...
}

// VS

match event.event_data {
    EventData::HistDlgState(data) =>...  // data.sn
}
```

Reply:
Hah, so you think this has to do with the fact that the struct is declared inline?

[ ... trying it out...]

You are right!

Reply:
The new match statement looks like this:

```rust

        match event.event_data {
            EventData::BuddyList(_event_data) => {
                // TODO
            }
            EventData::HistDlgState(_event_data) => {
                // TODO
            }
            EventData::MyInfo(_event_data) => {
                // TODO
                // purple_notify_userinfo
            }
            EventData::PermitDeny(_event_data) => {
                // TODO
            }
            EventData::Presence(_event_data) => {
                // TODO
            }
            EventData::GalleryNotify(_event_data) => {
                // TODO
            }
        }
```